### DISCUSSION TITLE
Running Renovate App on Windows agent?

### DISCUSSION THREAD

**Discussion by jessehouwing** [renovatebot/renovate#20159] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitHub

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

One of my projects has a windows-only dependency. Renovate fails to update the dependencies because of this.

I want it to work correctly ;).

### What you tried so far.

Nothing much yet. Not sure where to start.

### Relevant debug logs

<details><summary>Logs</summary>

```
npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for 7zip-bin-win@2.2.0: wanted {"os":"win32"} (current: {"os":"linux","arch":"x64"})
npm ERR! notsup Valid OS:    win32
npm ERR! notsup Valid Arch:  undefined
npm ERR! notsup Actual OS:   linux
npm ERR! notsup Actual Arch: x64

npm ERR! A complete log of this run can be found in:
npm ERR!     /tmp/renovate-cache/others/npm/_logs/2023-02-01T19_35_51_362Z-debug-0.log
```

https://github.com/jessehouwing/azure-devops-extension-tasks/pull/18

</details>

**Comment by viceice** [comment]
you need to run renovate self hosted and start it on windows.

**Reply by jessehouwing** [reply]
Was afraid of that

**Reply by viceice** [reply]
the hosted renovate app entirely runs on Linux. 🤷‍♂️

**Reply by jessehouwing** [reply]
Would be nice if for such purposes there would be a template Actions Workflow similar to how CodeQL is setup. That way I could 3-click-hurray! enable RenovateBot through my own GitHub Actions token for that repo. And the app could detect this repo is managed differently.

![image](https://user-images.githubusercontent.com/4173387/216284185-ded33da6-5ecf-47fc-931b-f9771d3cc2f4.png)

**Reply by rarkins** [reply]
Link to more details? We would like to have the concept of "self hosted runners" but for Renovate one day.

**Reply by jessehouwing** [reply]
You can send a pull request with the desired workflow file (and a few magic files that describe the workflow) to this repository: https://github.com/actions/starter-workflows

When it's been merged, the workflow will appear when people click "new workflow" in the actions tab in GitHub Actions. There is some magic templating you can do to set branch names and a few other things:
https://docs.github.com/en/actions/using-workflows/creating-starter-workflows-for-your-organization

You get to define the workflow file. Most workflow files inject a big comment block with "next steps" like setting a secret or setting up a github app.

**Reply by rarkins** [reply]
How is this different to what we have at https://github.com/renovatebot/github-action already?

**Reply by jessehouwing** [reply]
Well that relies on docker, and thus won't work on a runner that needs to run windows. 

Plus, I need to know where to find the action and how to setup the workflow correctly. Usually when you start from a new workflow and then pull the action from the marketplace, you copy/paste a single "step template", but it would be a strange place to overwrite a whole workflow.

That's where the templates come in, plus, the template could setup caching for the `cachedir` and a few other things correctly, pass the right parameters in the right way.

It took me 20 workflow runs to get the correct syntax to run Renovate against a single repository. The repo that hosts the workflow. To be honest, that would have been my expected behavior when running `npx renovate` inside a cloned git repo.

**Comment by jessehouwing** [comment]
It would augment. Basically making it easier to find and a 2 click getting
started.

**Reply by rarkins** [reply]
So there's nothing stopping you right now, as-is?

**Reply by jessehouwing** [reply]
Not to get running on the hosted runners, but from what i understand, the action relies on docker, so that still binds me to Linux only.

**Reply by viceice** [reply]
you can run `npx renovate ...` on a windows actions runner and it should do what you want.

**Reply by viceice** [reply]
you're probably also interested in following this issue:
- https://github.com/renovatebot/github-action/issues/571

**Reply by jessehouwing** [reply]
Looks like it :)

**Reply by jessehouwing** [reply]
I'd also like the action to detect it's running in an "issue" workflow and use the updated issue to run. That would allow me to click the checkbox on a renovate pull request and have renovate act on that.

I'm now experimenting with caching the `cachedir` folder and the `npm cache` to speed up renovate, cause it bloody takes forever on the hosted runner.

Snyk started distributing a single compiled exe for windows and a binary for linux so it could download that and run it without dependencies. The file is massive, but as it's hosted on a CDN, it's very quick to download to a hosted agent and run.

**Reply by rarkins** [reply]
I wouldn't mind bundling up Renovate (at least as a single Node file) but we should remove all other xGPL first before doing so.
